kurl_proxy/cmd/main.go

```diff
@@ -430,26 +434,25 @@ func getHttpsServer(upstream, dexUpstream *url.URL, tlsSecretName string, secret
 			}
 		}()
 	})
-	mux.Handle("/tls", r)
-	mux.Handle("/tls/", r)
```

Comment:
Why is this not needed anymore? Are both paths covered by the `r.GET("/tls", func(c *gin.Context)...` above?

Reply:
The various `/tls` routes were already registered with the gin Engine, however we didn't use the gin Engine as the handler in the http server that is returned from the function, we instead used a [separate ServeMux](https://github.com/replicatedhq/kots/pull/3725/files#diff-db27cebad99585565557ed130999dc35ca6e4d2e4cf648b04119580339d9d407L449) that was also created.  This introduced the need to register those `/tls` routes (that were already registered with the gin Engine) with the ServeMux, otherwise they would not be handled.  The change in this PR was to get rid of the additional ServeMux all together and use the [gin Engine as the handler](https://github.com/replicatedhq/kots/pull/3725/files#diff-db27cebad99585565557ed130999dc35ca6e4d2e4cf648b04119580339d9d407R445) for everything (static files, tls routes, and proxies to dex/kots).  The benefit of doing this as it pertains to [SC-70585](https://app.shortcut.com/replicated/story/70585/kots-clickjacking-security-vulnerability) is that we can [leverage gin's support for middleware functions](https://github.com/replicatedhq/kots/pull/3725/files#diff-db27cebad99585565557ed130999dc35ca6e4d2e4cf648b04119580339d9d407R285) to add headers to _all_ responses served.

Reply:
Okay I see, thank you. This looks good to me.